### PR TITLE
Create a placeholder for WASI threads implementation

### PR DESCRIPTION
**Pull request by loganek**
This a simpler version of the PR: https://github.com/bytecodealliance/wasm-micro-runtime/pull/1638
To simplify the review, I'll split it into a few smaller changes, starting from this one, which just creates a function stub and a simplistic example.

### CODE REVIEW

**Comment by yamt** [comment]
> This a simpler version of the PR: #1638 To simplify the review, I'll split it into a few smaller changes, starting from this one, which just creates a function stub and a simplistic example.

i feel the original version is simple enough.

**Comment by loganek** [comment]
@yamt it is indeed, but it's also a bit incorrect, and I'm not happy about the quality of that at all - I only made a PR to save my work and give others opportunity to provide higher-level feedback; I actually never intended to merge it as-it-is. As I came back to that work, I thought I'd use it as opportunity to split it into two separate PRs, where the first one is just to provide a template, and the second will be about the actual implementation (with some TODOs, which we address later on - that's why the development will happen on the `dev/wasi_threads` branch until we're fully ready). Having said that, many thanks for all your comments in the previous PR, I'll take them into account.

File: samples/wasi-threads/wasm-apps/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (wenyongh) [inline samples/wasi-threads/wasm-apps/CMakeLists.txt:1]: Should add the license header

File: samples/wasi-threads/wasm-apps/no_pthread.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#ifndef __wasi__
```
Comment (wenyongh) [inline samples/wasi-threads/wasm-apps/no_pthread.c:1]: Should add the license header.

**Review by wenyongh — APPROVED** [review]
LGTM

**Pull request merged**